src/child_process_tracker.rs

```diff
@@ -0,0 +1,160 @@
+use anyhow::Result;
+use std::sync::Arc;
+
+use tokio::process::Child;
+
+/// Windows does not have a concept of parent processes and so
+/// killing the deno task process will not also kill any spawned
+/// processes by default. To make this work, we can use winapi's
+/// jobs api, which allows for associating a main process so that
+/// when the main process terminates, it will also terminate the
+/// associated processes.
+///
+/// Read more: https://stackoverflow.com/questions/3342941/kill-child-process-when-parent-process-is-killed
+#[derive(Clone)]
+pub struct ChildProcessTracker(Arc<dyn Tracker>);
+
+impl ChildProcessTracker {
+  #[cfg(windows)]
+  pub fn new() -> Self {
+    match windows::WinChildProcessTracker::new() {
+      Ok(tracker) => Self(Arc::new(tracker)),
+      Err(err) => {
+        if cfg!(debug_assertions) {
+          panic!("Could not start tracking processes. {:#}", err);
+        } else {
+          // fallback to not tracking processes if this fails
+          Self(Arc::new(NullChildProcessTracker))
+        }
+      }
+    }
+  }
+
+  #[cfg(not(windows))]
+  pub fn new() -> Self {
+    Self(Arc::new(NullChildProcessTracker))
+  }
+
+  pub fn track(&self, child: &Child) {
+    if let Err(err) = self.0.track(child) {
+      if cfg!(debug_assertions) && child.id().is_some() {
+        panic!("Could not track process: {:#}", err);
+      }
+    }
+  }
+}
+
+trait Tracker: Send + Sync {
```

Comment:
I suppose this is a trait because you imagine there will be a unix implementation also?

(If not, consider calling the thing `JobObject` because that's really all it is)